authentik/sources/ldap/models.py

```diff
@@ -112,8 +123,20 @@ def server(self, **kwargs) -> Server:
         if self.peer_certificate:
             tls_kwargs["ca_certs_data"] = self.peer_certificate.certificate_data
             tls_kwargs["validate"] = CERT_REQUIRED
+        if self.client_certificate:
+            with NamedTemporaryFile(mode="w", delete=False) as temp_cert:
+                temp_cert.write(self.client_certificate.certificate_data)
+                certificate_file = temp_cert.name
+            with NamedTemporaryFile(mode="w", delete=False) as temp_key:
+                temp_key.write(self.client_certificate.key_data)
+                private_key_file = temp_key.name
+            tls_kwargs["local_private_key_file"] = private_key_file
+            tls_kwargs["local_certificate_file"] = certificate_file
+            tls_kwargs["validate"] = CERT_REQUIRED
         if ciphers := CONFIG.y("ldap.tls.ciphers", None):
             tls_kwargs["ciphers"] = ciphers.strip()
+        if self.sni:
+            tls_kwargs["sni"] = self.server_uri.strip()
```

Comment:
I honestly thought this was enabled by default, I don't think this will work in all cases as we allow multiple server URIs to be specified and separated by commas (See L146), so maybe this should instead be a text field?

Reply:
I think that's probably fair, assuming that most LDAP setups with more than 1 server have all of the servers IPs under a single DNS entry, and that DNS entry matches the certificate's CN or SAN